src/main/java/com/autotune/analyzer/recommendations/engine/DurationBasedRecommendationEngine.java

```diff
@@ -65,17 +69,18 @@ public AnalyzerConstants.RecommendationCategory getEngineCategory() {
 
     @Override
     public HashMap<String, Recommendation> getRecommendations(ContainerData containerData, Timestamp monitoringEndTime) {
-        // TODO: Needs to be implemented
-        AnalyzerConstants.RecommendationCategory recommendationCategory = AnalyzerConstants.RecommendationCategory.DURATION_BASED;
+        // Get the results
         HashMap<Timestamp, IntervalResults> resultsMap = containerData.getResults();
-        Timestamp minDate = resultsMap.keySet().stream().min(Timestamp::compareTo).get();
+        // Create a
         HashMap<String, Recommendation> resultRecommendation = new HashMap<String, Recommendation>();
-        for (RecommendationSubCategory recommendationSubCategory : recommendationCategory.getRecommendationSubCategories()) {
+        for (RecommendationSubCategory recommendationSubCategory : this.category.getRecommendationSubCategories()) {
             DurationBasedRecommendationSubCategory durationBasedRecommendationSubCategory = (DurationBasedRecommendationSubCategory) recommendationSubCategory;
             String recPeriod = durationBasedRecommendationSubCategory.getSubCategory();
             int days = durationBasedRecommendationSubCategory.getDuration();
-            Timestamp monitorStartDate = CommonUtils.addDays(monitoringEndTime, -1 * days);
-            if (monitorStartDate.compareTo(minDate) >= 0 || days == 1) {
+            Timestamp monitorStartDate = getMonitoringStartDate(resultsMap,
```

Comment:
`monitoringStartTime`

Reply:
Thanks @dinogun for the review, Made changes accordingly